contracts/utils/deploy.js

```diff
@@ -284,10 +284,29 @@ function deploymentWithProposal(opts, fn) {
       await sendProposal(propArgs, propDescription, propOpts);
       log("Proposal sent.");
     } else if (isFork) {
-      // On Fork we can send the proposal then impersonate the guardian to execute it.
-      log("Sending and executing proposal...");
-      await executeProposal(propArgs, propDescription, propOpts);
-      log("Proposal executed.");
+      let skipExecuteProposal = false;
+      if (proposalId) {
+        const proposalState = ["New", "Queue", "Expired", "Executed"][
+          await governor.state(proposalId)
+        ];
+        if (["New", "Queue"].includes(proposalState)) {
+          skipExecuteProposal = true;
+          console.log(
+            `Found proposal id: ${proposalId} on forked network. Executing proposal in place of deployment of: ${deployName}`
+          );
+          await executeProposalOnFork(proposalId);
+        }
+        governor.queue(id, { from: GOV_MULTISIG });
+        brownie.chain.sleep(48 * 60 * 60 + 1);
+        governor.execute(id, { from: GOV_MULTISIG });
```

Comment:
Does't `executeProposalOnFork()` do these (queue and execute) already? Won't they be called twice in such cases?

Reply:
committed those too early. deleting it, thanks